File: src/main/scala/components/updatecharacter/handlers/MockUpdateCharacterDataHandler.scala
package components.updatecharacter.handlers

import com.amazonaws.services.lambda.runtime.events.{APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent}
import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
import components.updatecharacter.repositories.MockCharacterUpdater

class MockUpdateCharacterDataHandler extends RequestHandler[APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent] {
  override def handleRequest(input: APIGatewayProxyRequestEvent, context: Context): APIGatewayProxyResponseEvent = {
    new UpdateCharacterDataHandler(MockCharacterUpdater)
      .handleRequest(input, context)
  }
}

File: src/main/java/components/updatecharacter/handlers/MockUpdateCharacterDataHandler.java
package components.updatecharacter.handlers;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
import components.updatecharacter.repositories.MockCharacterUpdater$;

@SuppressWarnings("unused")
public class MockUpdateCharacterDataHandler implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {

    @Override
    public APIGatewayProxyResponseEvent handleRequest(APIGatewayProxyRequestEvent input, Context context) {
        UpdateCharacterDataHandlerInternal mockUpdateCharacterDataHandler = new UpdateCharacterDataHandlerInternal(MockCharacterUpdater$.MODULE$);
        return mockUpdateCharacterDataHandler.handleRequest(input, context);
    }
}
